### ISSUE TITLE
write ForcingModel to netcdf instead of bc file

### ISSUE THREAD

**Issue by veenstrajelmer** [Deltares/HYDROLIB-core#318]
**Is your feature request related to a problem? Please describe.**
For DCSM, boundary conditions are converted from bc to netcdf since this saves 60% of the disk space. This is currently not available in hydrolib. Would it be possible to add a feature to write the ForcingModel to netcdf instead of a bc file? This would be more efficient than first writing a bc and then converting it.

**Additional context**
Currently, this script converts bc to netcdf:
https://repos.deltares.nl/repos/hydro_tools/data_conversion_scripts/bc2nc.m
Formatting is handled in this script:
https://repos.deltares.nl/repos/hydro_tools/data_conversion_scripts/writeBoundaryConditions2nc.m

**Comment by veenstrajelmer** [comment]
In hindsight, this might be a bit out of scope for hydrolib (since xarray), but it would be convenient if the ext file (pointing to the nc file) could be written with hydrolib anyway.

**Comment by veenstrajelmer** [comment]
Since we recently updated the dfm_tools ncbnd format to align with the FM/FEWS format, we are now close to using it in the modelbuilder (later hydromt_delft3dfm). 

An example where `file_nc_out` is a netcdf file:
```
boundary_object = hcdfm.Boundary(quantity=quantity,
                                 locationfile=file_pli,
                                 forcingfile=file_nc_out) #TODO: does not accept netcdf file
```

Raises:
```
ValidationError: 1 validation error for Boundary
forcingFile
  'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte (type=value_error.unicodedecode)
```

A solution would be to add a `DiskOnlyFileModel` as accepted input types.

**Comment by priscavdsluis** [comment]
> Since we recently updated the dfm_tools ncbnd format to align with the FM/FEWS format, we are now close to using it in the modelbuilder (later hydromt_delft3dfm).
> 
> An example where `file_nc_out` is a netcdf file:
> 
> ```
> boundary_object = hcdfm.Boundary(quantity=quantity,
>                                  locationfile=file_pli,
>                                  forcingfile=file_nc_out) #TODO: does not accept netcdf file
> ```
> 
> Raises:
> 
> ```
> ValidationError: 1 validation error for Boundary
> forcingFile
>   'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte (type=value_error.unicodedecode)
> ```
> 
> A solution would be to add a `DiskOnlyFileModel` as accepted input types.

@veenstrajelmer I think adding a `DiskOnlyFileModel` as supported types for the `forcingFile` is a good idea for now. 
 Do you still want to have netcdf reading/writing supported (in the future). If yes, I can create a new issue for your suggestion and keep this issue open.

**Comment by veenstrajelmer** [comment]
@priscavdsluis I agree with your suggestion to add a `DiskOnlyFileModel` as supported types for the `forcingFile`.

**Comment by priscavdsluis** [comment]
@rhutten discussed: we should make sure that we do not copy the files, but keep the path to the original file (if absolute).

**Comment by veenstrajelmer** [comment]
@priscavdsluis I agree, but also for relative paths I think. Would be good to discuss the possibility at least. Many models are set up such that they are moveable, hard coded absolute path are avoided as much as possible.

**Comment by veenstrajelmer** [comment]
First await https://github.com/Deltares/HYDROLIB-core/issues/553, then reconsider this issue.
